src/Kris/LaravelFormBuilder/FormBuilder.php

```diff
@@ -48,6 +50,15 @@ public function create($formClass, array $options = [], array $data = [])
             ->setFormOptions($options)
             ->addData($data);
 
+        // reset model to match named form logic
+        if ($form->getName()) {
```

Comment:
This logic can be moved maybe [here](https://github.com/kristijanhusak/laravel-form-builder/blob/master/src/Kris/LaravelFormBuilder/Form.php#L543).